.teamcity/hosted/scheduled/build/base/TeamCityScheduledImageBuildLinux_Base.kt

```diff
@@ -0,0 +1,107 @@
+ * The image uses parameter 'docker.nightlyRepository', which, for security reasons, should be defined at ...
+ * ... TeamCity-server level rather than DSL.
+ *
+ * @param platform target platform, e.g. amd64, aarch64 (ARM)
+ * @param agentReq a substring that agent name ('teamcity.agent.name') must contain
+ */
+class TeamCityScheduledImageBuildLinux_Base(private val platform: String, private val agentReq: String) : BuildType({
+    name = "[${platform}] TeamCity Docker Images - Automated Scheduled Build - Linux"
+    id("TeamCityScheduledImageBuildLinux_Base${platform}")
+
+    val images: Set<ImageInfo> = when {
+        platform.lowercase().contains("arm") || platform.lowercase()
+            .contains("aarch") -> ImageInfoRepository.getArmImages(
+            "%docker.nightlyRepository%",
+            "%dockerImage.teamcity.buildNumber%"
+        )
+
+        platform.lowercase().contains("amd") -> ImageInfoRepository.getAmdImages(
+            "%docker.nightlyRepository%",
+            "%dockerImage.teamcity.buildNumber%"
+        )
+
+        else -> throw IllegalArgumentException("Unable to find images for specified platform [${platform}]")
+    }
+
+    vcs {
+        root(TeamCityDockerImagesRepo.TeamCityDockerImagesRepo)
+    }
+
+    // all .yml files (e.g. compose samples)
+    artifactRules = "+:*.yml"
+
+    params {
+        // the images will be published into registry that holds nightly builds
+        param("docker.buildRepository", "%docker.nightlyRepository%")
+        // no postfix needed
+        param("docker.buildImagePostfix", "")
+        param("tc.image.version", "%dockerImage.teamcity.buildNumber%")
+    }
+
+    steps {
+        // build each image
+        images.forEach { imageInfo -> buildImage(imageInfo) }
+
+        // publish images if build of each one of them succeeded
+        images.forEach { imageInfo -> publishImage(imageInfo) }
+
+        script {
+            name = "Generate Sample docker-compose manifest for the created images"
+            scriptContent = """
+                cat <<EOF > teamcity-linux-%tc.image.version%.docker-compose.${platform.trim()}.yml
+                ${Utils.getSampleComposeFile("%docker.nightlyRepository%", "%tc.image.version%", "", platform)}
+                EOF
+                """.trimIndent()
+        }
+    }
+
+    dependencies {
+        dependency(AbsoluteId("TC2023_05_BuildDistDocker")) {
+            artifacts {
+                artifactRules = "TeamCity.zip!/**=>context/TeamCity"
+                cleanDestination = true
+                lastSuccessful()
+            }
+        }
+    }
+
+    features {
+        dockerSupport {
+            cleanupPushedImages = true
+            loginToRegistry = on {
+                dockerRegistryId = "PROJECT_EXT_315"
+            }
+        }
+    }
+
+    requirements {
+        // in case of aarch64, images must be built on aarch64-based agents
+        contains("teamcity.agent.name", agentReq)
```

Comment:
Shouldn't it be the same requirement as in https://github.com/JetBrains/teamcity-docker-images/pull/80/files#diff-231f2cba1dce79cc459764126e715f1073776c7dd0ae8ade1a83fb06fd498121R73 (or vise versa)?

Reply:
@burnasheva given that it's a base class for multiple build configurations, it depends on target image platform:
* ARM-based images - yes.
* AMD-based images - no, it could be empty.
We use it for both of them:
```
dependencies {
        arrayOf(
            TeamCityScheduledImageBuildLinux_Base("amd64", ""),
            TeamCityScheduledImageBuildLinux_Base("aarch64", "arm")
           )
            ….
}
```